services/src/test/scala/cromwell/pipeline/service/ProjectSearchServiceTest.scala

```diff
@@ -0,0 +1,104 @@
+  private val projectConfigurationId = ProjectConfigurationId.randomId
+
+  private val wdlParams: WdlParams =
+    WdlParams(Paths.get("/home/file"), List(FileParameter("nodeName", StringTyped(Some("hello")))))
+
+  private val activeConfiguration1: ProjectConfiguration =
+    ProjectConfiguration(
+      projectConfigurationId,
+      projectId,
+      active = true,
+      wdlParams,
+      ProjectConfigurationVersion.defaultVersion
+    )
+  private val activeConfiguration2: ProjectConfiguration =
+    ProjectConfiguration(
+      projectConfigurationId,
+      dummyProject2.projectId,
+      active = true,
+      wdlParams,
+      ProjectConfigurationVersion.defaultVersion
+    )
+
+  private val projectService = ProjectServiceTestImpl(dummyProjects: _*)
+  private val projectFileService = ProjectFileServiceTestImpl()
+  private val projectConfigurationService =
+    ProjectConfigurationServiceTestImpl(activeConfiguration1, activeConfiguration2)
+
+  private val projectSearch = ProjectSearchService(projectService, projectFileService, projectConfigurationService)
+
+  "ProjectSearchServiceTest" when {
+    "search for all owner projects" should {
+      "return list of all projects" in {
+        val filter = AllFilter
+        val request = ProjectSearchRequest(filter)
+
+        projectSearch.searchProjects(request, userId).map(_ shouldBe Seq(dummyProject1, dummyProject2, dummyProject3))
+      }
+    }
+
+    "search project by name full match" should {
+      "return full matched project" in {
+        val name = dummyProject1.name
+        val filter = ByNameFilter(FullMatch, name)
+        val request = ProjectSearchRequest(filter)
+
+        projectSearch.searchProjects(request, userId).map(_ shouldBe Seq(dummyProject1))
+      }
+    }
+
+    "search project by name regexp match" should {
+      "return matched project" in {
+        val regexString = ".*ir.*"
+        val filter = ByNameFilter(RegexpMatch, regexString)
+        val request = ProjectSearchRequest(filter)
+
+        projectSearch.searchProjects(request, userId).map(_ shouldBe Seq(dummyProject1, dummyProject3))
+      }
+    }
+
+    "search project with configs" should {
+      "return projects with config" in {
+        val filter = ByConfigFilter("exists", value = true)
+        val request = ProjectSearchRequest(filter)
+
+        projectSearch.searchProjects(request, userId).map(_ shouldBe Seq(dummyProject1, dummyProject2))
+      }
+    }
+
+    "search project with complex filter" should {
+      "return projects without config or name match" in {
+        val name = dummyProject1.name
+        val filterByName = ByNameFilter(FullMatch, name)
+        val filterByConfig = ByConfigFilter("exists", value = false)
+        val complexFilter = ComplexOrFilter(filterByName, filterByConfig)
+        val request = ProjectSearchRequest(complexFilter)
+
+        projectSearch.searchProjects(request, userId).map(_ shouldBe Seq(dummyProject1, dummyProject3))
+      }
```

Comment:
I'm seeing 6 different filter types and only 5 tests (2 of them are for `ByNameFilter`)  which means some filters are not covered by tests. Can you please the missing ones?
Also, filters like `ByConfigFilter` and `ByFileFilter` accept boolean parameter which means we should have at least two tests for each.

Reply:
There is a problem with current byFile search testing. Since files and projects are not connected by id it is impossible to filter files by project with current testImpl. So I tried to temporary solve that by making a new `ProjectFileServiceIdCombinedTestImpl` (masterful naming skills applied) which accepts  `(ProjectId, Seq[ProjectFile])` tuple. With that we can receive different search results for different projects during testing.

Reply:
I think it's fine to have a way to specify project id for project file, but I don't think it's a good idea to have several ways to create test implementation. You should create an overloaded version of `ProjectFileServiceTestImpl.apply` method that accepts `(ProjectId, ProjectFile)` instead of `ProjectFile`. Under the hood it might create another implementation, but it should still be single point of test implementation creation. And yeah, the name can be better I think.

Reply:
Or maybe we can modify existing `ProjectFileServiceTestImpl` to accept `(ProjectId, ProjectFile)` instead of `ProjectFile`, I think that is a valid choice too.

Reply:
`ProjectFileServiceTestImpl` modified to accept `(ProjectId, ProjectFile)`

Reply:
`ProjectFileServiceTestImpl` modified to accept `(ProjectId, ProjectFile)`